### ISSUE TITLE
Please Update 9anime Companion 

### ISSUE THREAD

**Issue by farasmughal** [lap00zza/9anime-Companion#143]
#### Select your issue type: (check at least one)
 
- [ ] Bug
- [ ] Question
- [ ] Suggestion
- [x] Other

#### Steps to reproduce:

1. Please Update Batch Download for RapidVideo 
2. Please Update Batch Download Streamango 

#### Describe your issue:
Please Update Batch Download for RapidVideo or Streamango I can't download without batch please update it

**Comment by Power27** [comment]
Or provide us with some update on when/if you might look at this.

Thanks.

**Comment by HRussellZFAC023** [comment]
*any* funtionality is better than none

**Comment by saineyjadama** [comment]
9 Anime is a great website

**Comment by antony-jr** [comment]
@farasmughal @Power27 @HRussell63250 @saineyjadama Why do you need 9Anime-Companion ? You can **block ads with UBlock Origin**. Or do you use 9anime-companion only to batch download ? Just something I'm curious about.

**Comment by HRussellZFAC023** [comment]
@antony-jr 
Back when it was working it was very cool. What I liked most about it remebered what I had watched and acted as a shortcut to the website. The global serach shortcut was also usefull when looking for an anime.
In general, id say i liked it because it was a pretty extension and it saved my time although there is nothing I cant do without it.

**Comment by antony-jr** [comment]
@HRussell63250 Got it ,So MyAnimeList Integration or Watch History was the most wanted. What do you think about batch downloads ?

**Comment by LexHDev** [comment]
Batch Downloads would be nice, HOWEVER, downloading all of the episodes for a series is a bit extreme and not what I would like to do with larger series. Being able to specify an episode range would be a nice feature. @antony-jr

**Comment by antony-jr** [comment]
@KuraiKokoro I'm really planning to do a similar project but not quite similar because a browser extension is weak... What I'm planning to do is to build an ultimate desktop app (In C++ , For something challenging).. You will have range downloads , batch downloads , break point streaming ( i.e at specific download you can just start watchiing , Just like watching online but the cache is stored in your disk ) and Integratiion with MyAnimeList like services... What I'm concerned about is that what I will be doing is literally illegal so I have no motivation. Any points you have that this can actually help people ?

**Comment by NatteeSetobol** [comment]
I want to get this to work so I'm currently investigating the token calculation. it seems like the number that is subtracted in the return is now being generated randomly? I have not clue. I also can' really find the source to the token calculation and such. I suspect it's all the bottom of the page source code. I'm going to try to debug if I can and hope for the best.  
BTW,I am not looking to download Animes to my computer, but rather stream it on my raspberry pi. If you have any info about how the token is generated, please share. If I find anything I will post it. Thanks.

**Comment by antony-jr** [comment]
@nsetobol I'm not really into token calculation since they will change it frequently and thats not very reliable... even their API is crap... So if we really want to download animes then we have to do webscraping because they don't change their style very often ( Even if they want to change it will take more time and I think they will think a lot about this type of change , API is very easy to change , If they just add a constant to the calculation and your software goes to hell ). What I'm interested and aiming to create is an AI that can do the copy paste work ( Yes it will need more resource just like a web browser but it will be reliable and easy to maintain , No I'm not talking about selenium type of stuff)

**Comment by debakarr** [comment]
@antony-jr @nsetobol It's all **in-vain** if you make your project public just like 9anime companion. I don't know whether you guys know about this or not, but there is this app, [anYme](https://github.com/zunjae/anYme), and it lets you do a lot more than 9anime comapanion. Also the main reason it works fine even now is because of few reasons:

* **Code is not public (it might using web scrapping for somethings it does)**
* **Collaboration with other people**
* **No desktop support (yes no windows/linux/mac)**

**Comment by antony-jr** [comment]
@Dibakarroy1997 Does it do batch downloads ?

**Comment by debakarr** [comment]
@antony-jr yes. Just look at the **tags** or [**features**](https://github.com/zunjae/anYme#features).

**Comment by antony-jr** [comment]
@Dibakarroy1997 I'm confused it says , "Look up pictures from your favorite show and batch-download them" , Does that mean it **batch-downloads pictures from myanimelist ?**. Do you use it , Did you try batch downloads ? If so did it download the anime or the pictures ?

**Comment by debakarr** [comment]
@antony-jr I remember using it a few years back and never have any problem downloading (now I have my own cruncyroll account, so didn't used it for quite a while). Still you can always download and try using it.

**Comment by LexHDev** [comment]
Afaik scrapers aren't illegal, so making a program that doesn't host the
content but accesses it is fine

**Comment by antony-jr** [comment]
@Dibakarroy1997 Just checked out the app , I don't know but it requires more third party apps like ADM. Not very user friendly , It seems like its made by a group of desperate anime fans... But I will test the app more.

**Comment by antony-jr** [comment]
@Dibakarroy1997 OMG!!! They just embeded 9anime's website in their app ??? And the worst part we can't batch download... You have to manually go through the episodes and click the download button which 9anime-campanion aims to solve....

**Comment by debakarr** [comment]
I would say if you want to just download batch anime, make your own script and keep it confined to a group of people and don't make it public. If there a way of sharing it without making it Open Source, then it's good. Because making something open source project without collaboration wouldn't last much.

**Comment by antony-jr** [comment]
@Dibakarroy1997 The fun part is when we do it public , Just wait and see...

**Comment by HRussellZFAC023** [comment]
https://nyaa.si/ is best way to download anime.

**Comment by debakarr** [comment]
@HRussell63250 ikr. Even better when you know how to integrate automation using rss.

**Comment by antony-jr** [comment]
@HRussell63250 You can only download until you have peers , What if you want to download a old anime that is good but no one else downloads it. Another problem is the **size** , its not encoded so you will download more than you want... But Yes its very reliable than other sources..

**Comment by NatteeSetobol** [comment]
@antony-jr I see, I'm not sure how I would just mine everything since there are a lot of Animes to mine. 

@Dibakarroy1997 The problem is that not all Animes work on myanimelist like Clannad.

I looked though the source code and still see where they actually do the calculation for the token. I think they might have changed the DD, I'm not sure whats happening but if I see the source code I would be able to see whats going on and also maybe keep it private so they won't change it again.

**Comment by debakarr** [comment]
@antony-jr use IRC then. There is also an option to automate everything. I remember using Ginpachi-Sensei. Also no need to worry about peers then. And AFAIK encoders have their own channels in IRC.
